### PR TITLE
Fix duplicate BotQuote submission on message interaction

### PR DESCRIPTION
**Pull request by CasBoom**
stops a second wjd response being sent by checking if that wjd already exists

### CODE REVIEW

File: app/Bots/Commands/QuoteCommand.php [hunk 1]
```diff
@@ -163,4 +168,9 @@ public function handle()
             'disable_notification' => true,
         ]);
     }
+
+    private function quoteDuplicate($quote)
+    {
+        return BotQuote::where('quote', $quote)->exists();
```
Comment (roelofr) [inline app/Bots/Commands/QuoteCommand.php:174]: While this definitely works, Gumbo users sometimes like to submit quotes in duplicate.

Maybe adding a `message_id` column to the `bot_quotes` table and using the `getMessageId` method might be better. Note that the `message_id` is a 64-bit Integer (see [Telegram Bot API > Message](https://core.telegram.org/bots/api#message)).

Alternatively , you can use the `getDate` method, which returns an Integer or DateTimeInterface (I don't know for sure), and check for a quote at or after that date.

Comment (roelofr) [reply app/Bots/Commands/QuoteCommand.php:174]: This does disable the feature to edit quotes. Maybe we could incorporate it in a later phase, if any demand for that is deemed usable.

Comment (CasBoom) [reply app/Bots/Commands/QuoteCommand.php:174]: oef, should definitely be reworked then.

Comment (roelofr) [reply app/Bots/Commands/QuoteCommand.php:174]: Editing quotes via an edited message is currently not supported, but it does trigger the bot again. It results in duplicates, so it's okay to "break" this behaviour (it's actually "bug riding" so no issue if it no longer works)

Comment (CasBoom) [reply app/Bots/Commands/QuoteCommand.php:175]: made it so that it looks at the message ID instead

File: app/Bots/Commands/QuoteCommand.php [hunk 2]
```diff
@@ -163,4 +169,9 @@ public function handle()
             'disable_notification' => true,
         ]);
     }
+
+    private function quoteDuplicate($quote)
```
Comment (roelofr) [inline app/Bots/Commands/QuoteCommand.php:173]: As CodeClimate says: unused private function.

File: database/migrations/2023_09_03_190145_add_message_id_to_bot_quotes.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('bot_quotes', function (Blueprint $table) {
+            //
```
Comment (roelofr) [inline database/migrations/2023_09_03_190145_add_message_id_to_bot_quotes.php:31]: Please revert the change too.

File: README.md [hunk 4]
```diff
@@ -51,3 +51,9 @@ Please have a look at the [Contribution Guide][contrib] on how to get started.
 [conscribo]: https://www.conscribo.nl/api/
 [tailwind]: https://tailwindcss.com
 [contrib]: ./CONTRIBUTING.md
+
+
+## testing the gumbot
+Via @BotFather you can create a new telegram bot
+Next, replace in the .env file the TELEGRAM_BOT_NAME and TELEGRAM_BOT_TOKEN varbiables with your botname and token.
+The command "php artisan bot:listen" will start polling the bot for updates.
```
Comment (roelofr) [inline README.md:59]:
```suggestion
The command `php artisan bot:listen` will start polling the bot for updates.
```

File: README.md [hunk 5]
```diff
@@ -51,3 +51,9 @@ Please have a look at the [Contribution Guide][contrib] on how to get started.
 [conscribo]: https://www.conscribo.nl/api/
 [tailwind]: https://tailwindcss.com
 [contrib]: ./CONTRIBUTING.md
+
+
+## testing the gumbot
+Via @BotFather you can create a new telegram bot
```
Comment (roelofr) [inline README.md:57]: Mentions on GitHub will poke and link a GitHub account, that's probably not intended.

```suggestion
Via [@BotFather](https://t.me/BotFather) you can create a new telegram bot
```

**Review by roelofr — CHANGES_REQUESTED** [review]

File: app/Bots/Commands/QuoteCommand.php (hunk 2 above)
Comment (CasBoom) [reply app/Bots/Commands/QuoteCommand.php:173]: removed

File: database/migrations/2023_09_03_190145_add_message_id_to_bot_quotes.php (hunk 3 above)
Comment (CasBoom) [reply database/migrations/2023_09_03_190145_add_message_id_to_bot_quotes.php:31]: reverted

**Review by roelofr — APPROVED** [review]
I fixed them quickly, if there's anyhing you'd like to alter, feel free!

**Comment by CasBoom** [comment]
> I fixed them quickly, if there's anyhing you'd like to alter, feel free!

looks good 👍
I think we could merge
